### ISSUE TITLE
Having problems loading images from App_Data since upgrading from v4.2.1 to v4.4.0

### ISSUE THREAD

**Issue by bkowski** [JimBobSquarePants/ImageProcessor#239]
Been using ImageProcessor.Web v4.2.1 to load images from App_Data and everything was working great. Decided to upgrade to v4.4.0 and images stopped loading completely.
Getting an object reference not set to an instance of an object error
I have not downloaded the ImageProcess.Web.Config package, all my ImageProcessor.Web settings are default installation settings.
I was using the following url to load these images:
https://localhost:44301/AppDataPartnersProvider.axd?5635801549130091039-Logo.png?width=224&height=56&bgcolor=fff

Along with the below custom IImageService implementation, not sure what I need to do to get this working with v4.4.0? 
Any help would be greatly appreciated:

```c#
using System;
using System.Collections.Generic;
using System.IO;
using System.Threading.Tasks;
using System.Web;
using System.Web.Hosting;
using ImageProcessor.Web.Services;

namespace Moneysoft.Web.Helpers
{
    public class AppDataPartnersImageService : IImageService
    {
        /// <summary>
        /// The prefix for the given implementation.
        /// </summary>
        private string prefix = "AppDataPartnersProvider.axd";

        /// <summary>
        /// Gets or sets the prefix for the given implementation.
        /// <remarks>
        /// This value is used as a prefix for any image requests that should use this service.
        /// </remarks>
        /// </summary>
        public string Prefix
        {
            get
            {
                return this.prefix;
            }

            set
            {
                this.prefix = value;
            }
        }

        /// <summary>
        /// Gets a value indicating whether the image service requests files from
        /// the locally based file system.
        /// </summary>
        public bool IsFileLocalService
        {
            get
            {
                return true;
            }
        }

        /// <summary>
        /// Gets or sets any additional settings required by the service.
        /// </summary>
        public Dictionary<string, string> Settings { get; set; }

        /// <summary>
        /// Gets or sets the white list of <see cref="System.Uri"/>. 
        /// </summary>
        public Uri[] WhiteList { get; set; }

        /// <summary>
        /// Gets a value indicating whether the current request passes sanitizing rules.
        /// </summary>
        /// <param name="path">
        /// The image path.
        /// </param>
        /// <returns>
        /// <c>True</c> if the request is valid; otherwise, <c>False</c>.
        /// </returns>
        public bool IsValidRequest(string path)
        {
            return true;
        }

        /// <summary>
        /// Gets the image using the given identifier.
        /// </summary>
        /// <param name="id">
        /// The value identifying the image to fetch.
        /// </param>
        /// <returns>
        /// The <see cref="System.Byte"/> array containing the image data.
        /// </returns>
        public async Task<byte[]> GetImage(object id)
        {
            const string AppData = "/App_Data/Partners";
            string imageRoot = HostingEnvironment.MapPath(AppData);

            if (imageRoot == null)
            {
                throw new HttpException(404, "No root path found to serve " + id);
            }

            // In this instance we are just processing a set path. 
            // If you are using the querystring params as a means of identifying the correct image
            // then you can do something with it here.
            string path = Path.Combine(imageRoot, id.ToString().Split(new[] { '&', '?' })[0]);
            byte[] buffer;

            // Check to see if the file exists.
            // ReSharper disable once AssignNullToNotNullAttribute
            FileInfo fileInfo = new FileInfo(path);

            if (!fileInfo.Exists)
            {
                throw new HttpException(404, "Nothing found at " + id);
            }

            using (FileStream file = new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read, 4096, true))
            {
                buffer = new byte[file.Length];
                await file.ReadAsync(buffer, 0, (int)file.Length);
            }

            return buffer;
        }
    }
}
```

**Comment by christopherbauer** [comment]
Can you use the debugger and step through the implementation and provide a line number?

**Comment by JimBobSquarePants** [comment]
Hi @bkowski, 

Thanks for providing the `IImageService` implementation, that was very helpful.

The error is being thrown as you haven't constructed an instance of the dictionary containing settings for the ImageModule to query. We had to introduce a bit of code for this version that queried the setting for a protocol value in some service instances to ensure we could correctly handle remote images. Unfortunately that broke your code. I've realised that in our internal implementations we don't do that for the LocalImagService so that serves as a poor example to copy from. I'll update that. 

Add this to your class and it should all get working again. 

        /// <summary>
        /// Initializes a new instance of the <see cref="AppDataPartnersImageService"/> class.
        /// </summary>
        public AppDataPartnersImageService()
        {
            this.Settings = new Dictionary<string, string>();
        }

@christopherbauer Cheers for stepping in.
